MaxText/tests/forward_pass_logit_checker.py

```diff
@@ -107,8 +107,9 @@ def main(config):
     )
 
     max_logging.log(f"{golden_logits[0] =}, {full_train_logits[0, 0, :]=}")
+    # compare first 4 token logits
     assert jax.numpy.allclose(
-            full_train_logits[0, :, :], golden_logits, rtol=1e-01, atol=1e-01, equal_nan=False
+            full_train_logits[0, :4, :], golden_logits[:4, :], rtol=float(config.rtol), atol=float(config.atol), equal_nan=False
```

Comment:
we shouldn't be hardcoding the number of tokens to compare, it should be coming from the number of tokens in the golden data

Reply:
Yeah, I agree. This may not be a good way. Let me change it back to all. But need to set loose limit. 

I am running some experiments [here](https://docs.google.com/spreadsheets/d/1o2RhkRpPXL5WD7FYb9kgm64qjuifOt0E0nfe3dc0YZE/edit?resourcekey=0-HGJ3wSqnzfMQ52O_Pf10sQ#gid=774077451), and absolute difference is quite large on TPU compared to CPU.

Reply:
I will need to add a token size arg to Mixtral test, otherwise the atol and rtol need to be huge. It does not make sense to have a test with big value of those tolerance. Currently, Mixtral has 11 tokens due to bos and eos.